tests/attr/test_integrated_gradients_basic.py

```diff
@@ -253,86 +264,101 @@ def _assert_batched_tensor_multi_input(
 
     def _compute_attribution_and_evaluate(
         self,
-        model,
-        inputs,
-        baselines=None,
-        target=None,
-        additional_forward_args=None,
-        type="vanilla",
-        approximation_method="gausslegendre",
-    ):
+        model: nn.Module,
+        inputs: TensorOrTupleOfTensors,
+        baselines: Optional[Any] = None,
+        target: Optional[int] = None,
+        additional_forward_args: Optional[Any] = None,
+        type: Optional[str] = "vanilla",
+        approximation_method: str = "gausslegendre",
+    ) -> Tuple[Tensor, ...]:
         r"""
             attrib_type: 'vanilla', 'smoothgrad', 'smoothgrad_sq', 'vargrad'
         """
         ig = IntegratedGradients(model)
-        if not isinstance(inputs, tuple):
-            inputs = (inputs,)
-
-        if baselines is not None and not isinstance(baselines, tuple):
-            baselines = (baselines,)
+        inputs_tuple: Tuple[Tensor, ...]
+        if isinstance(inputs, Tensor):
+            inputs_tuple = (inputs,)
+        else:
+            inputs_tuple = inputs
 
+        baselines_tuple: Tuple[Any, ...]
         if baselines is None:
-            baselines = _tensorize_baseline(inputs, _zeros(inputs))
+            baselines_tuple = _tensorize_baseline(inputs_tuple, _zeros(inputs_tuple))
+        elif not isinstance(baselines, tuple):
+            baselines_tuple = (baselines,)
+        else:
+            baselines_tuple = baselines
 
+        attributions: Tuple[Tensor, ...]
```

Comment:
I think for this issue, it might be cleaner to avoid declaring the types in advance here and just use "cast" to cast the return value to the expected type. For instance, below, it would be something like:
`attributions, delta = cast(Tuple[Tuple[Tensor, ...], Tensor], ig.attribute ...`

You'll need to import cast from typing, it won't have any effect on runtime, just for type hinting.

Reply:
Casting was what I was missing to make life easier, when we "know" the right thing. I think with that, the latest changes are much more straight forward.